### DISCUSSION TITLE
Cannot upload to GitHub release

### DISCUSSION THREAD

**Discussion by qouteall** [Kira-NT/mc-publish#84] [Q&A] [answered]
It can upload to CurseForge and Modrinth but uploading to GitHub Release is broken

```
Error: HttpError: 403 (Forbidden, {"message":"Resource not accessible by integration","request_id":"6440:34AC:11F345E:14FF415:64BBC120","documentation_url":"https://docs.github.com/rest"})
```

https://github.com/iPortalTeam/PortalGun/actions/runs/5630448701/job/15256447219

Is it an issue of GitHub or my configuration?

**Comment by Kira-NT** [comment, marked as answer]
Hi! Just to clarify, this issue doesn't stem from either your actions or the ones of `mc-publish`. Rather, it's a result of GitHub altering the default workflow permission set from read & write to read-only. This modification has caused many features that previously functioned seamlessly in older repositories to malfunction, leaving numerous developers fairly confused (myself included).

The solutions to this problem have already been discussed in the pinned issue #55. However, for the sake of your convenience, I'll summarize them here as well. Essentially, you have two options:

1) Assign the necessary permissions to a specific job. In this case, since `mc-publish` attempts to create a release, it requires `contents: write` permission. You can specify this as follows:

    ```yaml
    jobs:
      build:
    
        runs-on: ubuntu-latest
    
        permissions:
            contents: write
    
        steps:
          # ...
    ```

   Further details about this feature and its syntax can be found [here](https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs).

2) As an alternative, you can revert to the previous default permission set for workflows within your repository's settings. To do this, select `Read and write permissions` under `Settings -> Actions -> General -> Workflow permissions`:

    ![Permissions](https://user-images.githubusercontent.com/40232067/234666530-35d7293c-0d4d-4e2c-89e0-50985614623c.png)

I hope this helps! Please, let me know if the issue persists!

As a side note, I believe I've found the next mod I'll be adding to my world *(whenever I manage to find time to play Minecraft again..)* xD Somehow I'd never considered searching for a Portal-like mod for Fabric before

**Comment by qouteall** [comment]
It worked. Thanks.

This should be mentioned in readme

**Reply by Kira-NT** [reply]
Oh, absolutely! It will be a part of README and the upcoming wiki :)
